Demo/Client/src/hooks/useTldrawStore.ts

```diff
@@ -0,0 +1,343 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import {
+    InstancePresenceRecordType,
+    SerializedSchema,
+    TLAnyShapeUtilConstructor,
+    TLInstancePresence,
+    TLRecord,
+    TLStoreWithStatus,
+    compareSchemas,
+    computed,
+    createPresenceStateDerivation,
+    createTLStore,
+    defaultShapeUtils,
+    defaultUserPreferences,
+    getUserPreferences,
+    react,
+    setUserPreferences,
+} from '@tldraw/tldraw';
+import { useEffect, useMemo, useState } from 'react';
+import { YKeyValue } from 'y-utility/y-keyvalue';
+import * as Y from 'yjs';
+import { useYjs } from './useYjs';
+
+export type UseTldrawStoreOptions = {
+    id: string;
+    shapeUtils?: TLAnyShapeUtilConstructor[];
+};
+
+export function useYjsTldrawStore(opts: UseTldrawStoreOptions) {
+    const { yjsDocument, yjsConnector } = useYjs();
+    const [store] = useState(() => {
+        const store = createTLStore({
+            shapeUtils: [...defaultShapeUtils, ...(opts?.shapeUtils || [])],
+        });
+
+        return store;
+    });
+
+    const [storeWithStatus, setStoreWithStatus] = useState<TLStoreWithStatus>({
+        status: 'loading',
+    });
+
+    const { yDoc, yStore, meta, room } = useMemo(() => {
+        const yArr = yjsDocument.getArray<{ key: string; val: TLRecord }>(
+            `tl_${opts.id}`
+        );
+        const yStore = new YKeyValue(yArr);
+        const meta = yjsDocument.getMap<SerializedSchema>('meta');
+
+        return {
+            yDoc: yjsDocument,
+            yStore,
+            meta,
+            room: yjsConnector,
+        };
+    }, [opts, yjsConnector, yjsDocument]);
+
+    useEffect(() => {
+        setStoreWithStatus({ status: 'loading' });
+
+        const unsubs: (() => void)[] = [];
+
+        function handleSync() {
+            // 1.
+            // Connect store to yjs store and vis versa, for both the document and awareness
+
+            /* -------------------- Document -------------------- */
+
+            // Sync store changes to the yjs doc
+            unsubs.push(
```

Comment:
Is it possible to refactor this into smaller methods. It is pretty long.

Reply:
I can try. I just copied it from tldraw's [example](https://github.com/tldraw/tldraw-yjs-example/blob/main/src/useYjsStore.ts),